src/Shared/TypeUtilities.cs

```diff
@@ -21,56 +21,53 @@ namespace Microsoft.Build.Shared
     internal static class TypeUtilities
     {
         /// <summary>
-        /// Determines whether the specified <paramref name="type"/> is decorated with an attribute of type <typeparamref name="T"/>.
+        /// Gets all attributes decorating the specified <paramref name="type"/>.
         /// </summary>
-        public static bool HasAttribute<T>(this Type type)
-            where T : Attribute
-        {
-            return type.HasAttribute(typeof(T).Name);
-        }
-
-        /// <summary>
-        /// Determines whether the specified <paramref name="type"/> is decorated with an attribute whose simple type name
-        /// equals <paramref name="attributeName"/>.
-        /// </summary>
-        public static bool HasAttribute(this Type type, string attributeName)
+        public static IList<CustomAttributeData> GetCustomAttributes(this Type type)
```

Comment:
FYI this is to avoid a double list + attribute allocation that happens since `LoadedType` always has to check for two attributes.